tests/conftest.py

```diff
@@ -160,55 +162,58 @@ def data_client(client):
 
 @fixture
 def dummy_response():
-    return {
-        "_shards": {"failed": 0, "successful": 10, "total": 10},
-        "hits": {
-            "hits": [
-                {
-                    "_index": "test-index",
-                    "_type": "company",
-                    "_id": "elasticsearch",
-                    "_score": 12.0,
-                    "_source": {"city": "Amsterdam", "name": "Elasticsearch"},
-                },
-                {
-                    "_index": "test-index",
-                    "_type": "employee",
-                    "_id": "42",
-                    "_score": 11.123,
-                    "_routing": "elasticsearch",
-                    "_source": {
-                        "name": {"first": "Shay", "last": "Bannon"},
-                        "lang": "java",
-                        "twitter": "kimchy",
+    return ObjectApiResponse(
```

Comment:
In elasticsearch-py 8.x, the response pretends it is a dictionary to be backwards-compatible, but it does not support `__setitem__` that is needed for faceted search to work. This is why we explicitly request `.body` in `Search.execute`: this is an actual dictionary that supports `__setitem__`.

The alternative is to add `__setitem__` support in elastic-transport-python.